tests/cpp/generate_config.cpp

```diff
@@ -101,3 +101,57 @@ TEST(GenerationConfigTest, valid_frequency_penalty) {
     config.frequency_penalty = -2.0;
     EXPECT_NO_THROW(config.validate());
 }
+
+ov::genai::GenerationConfig speculative_decoding_multinomial() {
+    auto speculative_decoding_multinomial_config = ov::genai::multinomial();
+    speculative_decoding_multinomial_config.num_assistant_tokens_schedule = ov::genai::NumAssistatantTokensScheduleType::CONSTANT;
+    speculative_decoding_multinomial_config.num_assistant_tokens = 5;
+    return speculative_decoding_multinomial_config;
+}
+
+ov::genai::GenerationConfig speculative_decoding_greedy() {
+    auto speculative_decoding_greedy_config = ov::genai::greedy();
+    speculative_decoding_greedy_config.num_assistant_tokens_schedule = ov::genai::NumAssistatantTokensScheduleType::HEURISTIC;
+    speculative_decoding_greedy_config.assistant_confidence_threshold = 0.4f;
+    return speculative_decoding_greedy_config;
+}
+
+TEST(GenerationConfigTest, invalid_static_spec_decoding) {
+    GenerationConfig config = speculative_decoding_greedy();
+    config.num_assistant_tokens_schedule = NumAssistatantTokensScheduleType::CONSTANT;
+    config.num_assistant_tokens = 0;
+    config.assistant_confidence_threshold = 0.5;
+    EXPECT_THROW(config.validate(), ov::Exception);
+    config.num_assistant_tokens_schedule = NumAssistatantTokensScheduleType::HEURISTIC;
+    config.num_assistant_tokens = 5;
+    config.assistant_confidence_threshold = 0;
+    EXPECT_THROW(config.validate(), ov::Exception);
+}
+
+TEST(GenerationConfigTest, valid_static_spec_decoding) {
+    GenerationConfig config = speculative_decoding_greedy();
+    config.num_assistant_tokens_schedule = NumAssistatantTokensScheduleType::CONSTANT;
+    config.num_assistant_tokens = 5;
+    config.assistant_confidence_threshold = 0;
+    EXPECT_NO_THROW(config.validate());
+}
+
+TEST(GenerationConfigTest, invalid_dynamic_spec_decoding) {
+    GenerationConfig config = speculative_decoding_greedy();
+    config.num_assistant_tokens_schedule = NumAssistatantTokensScheduleType::HEURISTIC;
+    config.num_assistant_tokens = 5;
+    config.assistant_confidence_threshold = 0.5;
+    EXPECT_THROW(config.validate(), ov::Exception);
+    config.num_assistant_tokens_schedule = NumAssistatantTokensScheduleType::CONSTANT;
+    config.num_assistant_tokens = 0;
+    config.assistant_confidence_threshold = 0.5;
+    EXPECT_THROW(config.validate(), ov::Exception);
+}
+
```

Comment:
do we have tests to check output values and compare with expected ones?

Reply:
@rkazants I so confused by the comment. Check the project sources

Reply:
Could you please point out where we check output values for different speculative decoding? To my understanding, it should be in this PR. If something new is implemented, new tests are implemented as well or tests previously implemented are enabled here.
Now I see that we check that no exceptions happen.

Reply:
@iefode, looks missed comment. Please check.